### PR TITLE
[6X] Detect client disconnection while running query and interrupt execution

### PR DESCRIPTION
**Pull request by maksm90**
Backport https://github.com/greenplum-db/gpdb/pull/12223 to 6X_STABLE

### CODE REVIEW

File: src/include/utils/sync_guc_name.h [hunk 1]
```diff
@@ -1,5 +1,6 @@
 		"bytea_output",
 		"check_function_bodies",
+		"client_connection_check_interval",
```
Comment (interma) [inline src/include/utils/sync_guc_name.h:3]: the master commit places this guc in [unsync_guc_name.h](https://github.com/greenplum-db/gpdb/pull/12223/files#diff-93ef61f5c972dff1ecd8f223a3f26cdb5cb0d2263a3e82ba057016fcdd6ce5a3)
I think the master pr is right? (no need to sync to QE)

Comment (maksm90) [reply src/include/utils/sync_guc_name.h]: Thanks, moved to unsync_guc_name.h

**Comment by interma** [comment]
thanks maksm90  for your contribution!

**Comment by interma** [comment]
I will add a test case for this PR later.

**Comment by interma** [comment]
@maksm90 I have added test case (and passed concourse), but looks no permission to push to your PR.

Would you help to pick it to this PR?
https://github.com/interma/gpdb/commit/821d0738b8cf4e7fee3d0caa942875f2add52b9e

Thanks.

**Comment by maksm90** [comment]
> Would you help to pick it to this PR?

Sure

> @maksm90 I have added test case (and passed concourse)

Great! I haven't supposed that reproducing of this kind of client failure is possible within isolation test framework.

**Pull request merged**